pennylane_qiskit/basic_aer.py

```diff
@@ -54,14 +57,56 @@ class BasicAerDevice(QiskitDevice):
 
     def __init__(self, wires, shots=1024, backend="qasm_simulator", **kwargs):
 
-        max_ver = Version("0.45.3")
+        max_ver = Version("0.46", partial=True)
 
         if Version(qiskit.__version__) > max_ver:
             raise RuntimeError(
-                f"The devices in the PennyLane Qiskit plugin are currently only compatible "
-                f"with versions of Qiskit below 0.46. You have version {qiskit.__version__} "
-                f"installed. Please downgrade Qiskit to use the devices. The devices will be "
-                f"updated in the coming weeks to be compatible with Qiskit 1.0!"
+                f"Qiskit has discontinued the BasicAer device, so it can only be used in"
+                f"versions of Qiskit below 1.0. You have version {qiskit.__version__} "
+                f"installed. For a Python simulator, use the 'qiskit.basicsim' device "
+                f"instead. Alternatively, you can downgrade Qiskit to use the "
+                f"'qiskit.basicaer' device."
             )
 
         super().__init__(wires, provider=qiskit.BasicAer, backend=backend, shots=shots, **kwargs)
+
+
+class BasicSimulatorDevice(QiskitDevice):
+    """A PennyLane device for the native Python Qiskit simulator from BasicProvider.
+
+    For more information on the BasicSimulator backend options and transpile options, please visit the
+    `BasicProvider documentation <https://docs.quantum.ibm.com/api/qiskit/providers_basic_provider>`_.
+
+    A range of :code:`backend_options` that will be passed to the simulator and
+    a range of transpile options can be given as kwargs.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['aux_wire', 'q1', 'q2']``).
+        backend (str): the desired backend
+        shots (int or None): number of circuit evaluations/random samples used
+            to estimate expectation values and variances of observables. For statevector backends,
+            setting to ``None`` results in computing statistics like expectation values and variances analytically.
```

Comment:
Is ``None`` an option here?

Reply:
Interesting question. The PennyLane device rejects that as an option because this isn't included in the list of state-based simulators. I'm also not sure if it would work if we treat it as a state-based simulator with the current UI. They've added a new way of dealing with state vectors, as in this example:

```
from qiskit import QuantumCircuit
 
qc = QuantumCircuit(3)
qc.h(0)
qc.h(1)
qc.cx(1, 2)
qc.measure_all()
 
# Previous
from qiskit import BasicAer
backend = BasicAer.get_backend("statevector_simulator")
statevector = backend.run(qc).result().get_statevector()
 
# Current
qc.remove_final_measurements()  # no measurements allowed
from qiskit.quantum_info import Statevector
statevector = Statevector(qc)
```

and our devices currently rely on the `backend.run(qc).result().get_statevector()`, which still exists for the backend we use in our tests and examples. So basically this backend allows having no shots, but implementing it would be adding new functionality, because it doesn't allow no shots via the old UI that we rely on currently. All the existing devices with state-based measurements will (for now) continue to work (though many of them are deprecated). This device just doesn't support it even though the backend could, because the plugin isn't set up to handle it.

Reply:
Currently `dev = qml.device("qiskit.basicsim", wires=5, shots=None)` creates a device with 1024 shots, and displays `UserWarning: The analytic calculation of expectations, variances and probabilities is only supported on statevector backends, not on the basic_simulator. Such statistics obtained from this device are estimates based on samples.`

I could add a custom warning to this device, since technically this is not an issue with the backend.

Expanding the functionality will be, I think, easier on the new device prototype, and I would say it part of this shortcut story: https://app.shortcut.com/xanaduai/story/54397/support-for-state-based-simulation-is-added-to-a-separate-device-that-inherits-from-the-qiskitdevice2-base-device

Reply:
Here are the associated Qiskit docs where I got the example above: https://docs.quantum.ibm.com/api/migration-guides/qiskit-1.0-features#basicaer-1

Reply:
> I could add a custom warning to this device, since technically this is not an issue with the backend.

Added a custom warning if None is passed, saying that the plugin doesn't support this at the moment (since its not accurate that the backend doesn't support it, as stated in the normal error message).